src/index.test.ts

```diff
@@ -223,3 +223,16 @@ it('assertGoError works', () => {
   const err = res.error;
   expect(err).toBe(err);
 });
+
+// This problem is caused because the value returned by go utils is an array with additional object properties which is
+// difficult to serialize and test equality.
+it('show testing limitation', () => {
```

Comment:
I still like the tuple syntax 😢 This is also a very weird way of using the function

But it does seem like it provides much better TS support so yeah this is probably a good idea

Reply:
But this is only a problem for the testing, isn't it? Nobody will actually compare the resulting object/tuple.

That being said, do we really need two ways of providing a result? We're not building a general-purpose library so everyone can use it. We're building it primarily for our internal needs and I think we will stick to the one format anyway. I agree it's a nice touch but I don't think it's really needed tbh.

Reply:
It also provides strange autocompletion, because of the mixed array (tuple) and object. See:
<img width="949" alt="image" src="https://user-images.githubusercontent.com/22679154/155297897-853625c9-10eb-4878-bcce-a05e1b4fec7c.png">

All array methods are there and you need to search for `succ` for the autocompletion to suggest the property for you. I'd say let's use the object syntax only.